src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -135,9 +139,50 @@ public Optional<Address> getRetiringFederationAddress() {
         return federatorSupport.getRetiringFederationAddress();
     }
 
+    @Override
+    public Optional<Federation> getProposedFederation() {
+        if (!federatorSupport.getConfigForBestBlock().isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+       
+        Optional<Address> proposedFederationAddress = getProposedFederationAddress();
+        Integer federationSize = federatorSupport.getProposedFederationSize()
```

Comment:
I think you don't need the size to check if a proposedFederation exists.

Reply:
I agree, but since this is also being done in the retiring federation impl I left it as is. Do you recommend removing it?

Reply:
actually, since you are using the fed size below, i think we could just check for it and remove the address check

Reply:
Alright I will delete the check for the address part then